api/src/test/java/org/openmrs/util/ConceptReferenceRangeUtilityTest.java

```diff
@@ -0,0 +1,292 @@
+		person.setBirthdate(calendar.getTime());
+		person.setGender("M");
+		String criteria = "$patient.getAge() > 1 && $patient.getAge() < 10 && $patient.getGender().equals('M')";
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeAndGenderMatch_shouldReturnFalseIfAgeDoesNotMatch() {
+		calendar.add(Calendar.YEAR, -11);
+		person.setBirthdate(calendar.getTime());
+		person.setGender("M");
+		String criteria = "$patient.getAge() > 1 && $patient.getAge() < 10 && $patient.getGender().equals('M')";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeAndGenderMatch_shouldReturnFalseIfGenderDoesNotMatch() {
+		calendar.add(Calendar.YEAR, -5);
+		person.setBirthdate(calendar.getTime());
+		person.setGender("F");
+		String criteria = "$patient.getAge() > 1 && $patient.getAge() < 10 && $patient.getGender().equals('M')";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeAndGenderMatch_shouldReturnFalseIfPersonIsNull() {
+		String criteria = "$patient.getAge() > 1 && $patient.getAge() < 10 && $patient.getGender().equals('M')";
+
+		try {
+			ConceptReferenceRangeUtility.evaluateCriteria(criteria, null);
+		} catch (ValidationException e) {
+			Assertions.assertEquals("Failed to validate with reason: patient is null", e.getMessage());
+		}
+	}
+
+	@Test
+	public void testAgeAndGenderMatch_shouldReturnFalseIfAgeOrGenderIsNull() {
+		String criteria = "$patient.getAge() > 1 && $patient.getAge() < 10 && $patient.getGender().equals('M')";
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testObsValueMatch_shouldReturnTrueIfValueTextMatch() {
+		person.setGender("F");
+		
+		Obs obs = buildObservation();
+		obs.setPerson(person);
+		obs.setValueText("PREGNANT");
+
+		Mockito.when(obsService.getLatestObsByConceptId("5089")).thenReturn(obs);
+
+		String criteria = "$patient.getGender().equals('F') && $fn.getLatestObsByConceptId('5089').getValueText().equals('PREGNANT')";
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testObsValueMatch_shouldReturnFalseIfValueMismatch() {
+		person.setGender("M");
+
+		String criteria = "$patient.getGender().equals('F') && $fn.getLatestObsByConceptId('5089') == true";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testTimeOfDay_shouldReturnTrueIfTimeMatches() {
+		int hour = LocalTime.now().getHour();
+		String criteria = "$fn.getTimeOfTheDay() == " + hour + "";
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	private Obs buildObservation() {
+		Concept concept = new Concept(5089);
+		concept.setDatatype(new ConceptDatatype(3));
+		
+		Encounter encounter = new Encounter(3);
+		Date datetime = new Date();
```

Comment:
What is the need for declaring all these variables, if all you use them for is the setters below?